service/controller/resource/scrapeconfigs/resource.go

```diff
@@ -83,14 +89,33 @@ func toSecret(v interface{}, provider string) (*corev1.Secret, error) {
 	return scrapeConfigsSecret, nil
 }
 
-func renderTemplate(clusterID string, provider string) (string, error) {
-	var templateData = TemplateData{
+func getTemplateData(cluster metav1.Object, provider string, clients k8sclient.Interface) (*TemplateData, error) {
+	c, ok := cluster.(*v1alpha2.Cluster)
+	if !ok {
+		return nil, microerror.Mask(fmt.Errorf("invalid v1alpha2.Cluster %T", cluster))
+	}
+
+	infra, err := clients.G8sClient().InfrastructureV1alpha2().AWSClusters(c.Spec.InfrastructureRef.Namespace).Get(c.Spec.InfrastructureRef.Name, metav1.GetOptions{})
```

Comment:
Does that not only make it usable on AWS Control planes?

Reply:
Right, this make it only usable with latest cluster using cluster api. But I've added legacy cluster already, see https://github.com/giantswarm/prometheus-meta-operator/pull/81/files#diff-251764767bca0bc62afcb4201ca2d51cR95-R107

Reply:
Nice. A question from me, can the etcd host name change once for a tenant cluster or can we inject it in the config?

Reply:
I don't think the etched host name ever changes. Where in the config do you want to inject that?

Reply:
Well, maybe not the config but I was thinking we could extract the get infra stuff in a function outside getTemplateData and pass the infra to getTemplateData instead to avoid bloating the function. I am not sure where we could put it though

Reply:
I honestly also don't know where to put this logic, I think it causes no harm here.